### PR TITLE
fix: fix: neo inferentia as compilation target not using framework ver

### PR DESCRIPTION
**Pull request by HappyAmazonian**
…sion

*Issue #, if available:*

*Description of changes:* For inferentia as compilation target device, we need to add Framework Version in the create_compilation_job requests to get the best result.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/model.py [hunk 1]
```diff
@@ -728,13 +728,22 @@ def _compilation_job_config(
...
+            target_instance_type: str, framework: str, framework_version: str
         ):
+            if target_instance_type and framework and framework_version:
+                framework = framework.lower()
+                multi_version_frameworks_support_mapping = {
+                    "ml_inf": ["pytorch", "tensorflow", "mxnet"],
+                    "ml_ioc": ["pytorch", "tensorflow"],
+                }
+                if re.match("(?=^ml_)", target_instance_type):
+                    return framework in multi_version_frameworks_support_mapping["ml_ioc"]
```
Comment (navinsoni) [inline src/sagemaker/model.py:741]: what are the options for instance type. do they have ml_ioc.

Comment (HappyAmazonian) [reply src/sagemaker/model.py:741]: ioc target means: ml_c4, ml_c5, ml_m4, ml_m5, ml_p2, ml_p3, ml_g4dn, they support torch and tf framework_versions.
ml_inf1 supports torch, tf and mxnet.

I can make it more explicit if you think it is better. (I was trying to make the change minimal)

Comment (navinsoni) [reply src/sagemaker/model.py:741]: do you want to select ioc first and then inf

**Review by navinsoni — APPROVED** [review]

File: src/sagemaker/model.py [hunk 2]
```diff
@@ -728,13 +729,22 @@ def _compilation_job_config(
...
-            framework.lower() in multiple_version_supported_framework_list
-            and target_instance_type is not None
-            and re.match("(?=^ml_)(?!ml_inf)", target_instance_type) is not None
-            and framework_version is not None
+        def multi_version_compilation_supported(
+            target_instance_type: str, framework: str, framework_version: str
         ):
+            if target_instance_type and framework and framework_version:
+                framework = framework.lower()
+                multi_version_frameworks_support_mapping = {
```
Comment (lekk32) [inline src/sagemaker/model.py:737]: This is fine for cloud targets but we also support framework versions for some edge targets now. Will need to fix that too in future.

**Review by lekk32 — APPROVED** [review]
I recall there was another bug for edge targets not getting framework version, thats not being handled here right?

**Review by mufaddal-rohawala — APPROVED** [review]

**Pull request merged**